eZ/Publish/Core/Repository/Helper/NameSchemaService.php

```diff
@@ -237,7 +237,8 @@ protected function getFieldTitles(array $schemaIdentifiers, $contentType, array
                     foreach ($contentType->fieldDefinitions as $spiFieldDefinition) {
                         if ($spiFieldDefinition->identifier === $fieldDefinitionIdentifier) {
                             $fieldDefinition = $this->contentTypeDomainMapper->buildFieldDefinitionDomainObject(
-                                $spiFieldDefinition
+                                $spiFieldDefinition,
+                                $languageCode
```

Comment:
> So, to get actual mainLanguageCode here it would require injecting Persistence Language Handler into NameSchemaService and retrieve it via $languageHandler->load($contentType->initialLanguageId)->languageCode which, I guess, wouldn't be efficient.

Nope.


Maybe just `key($spiFieldDefinition->name) ?: null`